### PR TITLE
Add ReprC impls for other array lengths and booleans

### PR DESCRIPTION
**Pull request by lionel1704**

### CODE REVIEW

File: src/repr_c.rs [hunk 1]
```diff
@@ -120,3 +120,30 @@ impl ReprC for [u8; 64] {
         Ok(*repr_c)
     }
 }
+
+impl ReprC for [u8; 48] {
```
Comment (m-cat) [inline src/repr_c.rs]: Minor point, but `48` is out of order, can it be put between `32` and `64`?

Comment (m-cat) [reply src/repr_c.rs]: Can you add a TODO comment to replace these with a const generic implementation, once that is stable?

File: src/repr_c.rs [hunk 2]
```diff
@@ -120,3 +120,30 @@ impl ReprC for [u8; 64] {
...
+impl ReprC for [u8; 96] {
+    type C = *const [u8; 96];
+    type Error = ();
+
+    unsafe fn clone_from_repr_c(repr_c: Self::C) -> Result<Self, Self::Error> {
+        Ok(*repr_c)
+    }
+}
+
+impl ReprC for bool {
```
Comment (m-cat) [inline src/repr_c.rs]: In SCL until now we have been using `u32` to denote bools. Any reason to use `i32` here?

Also, can you please update the comment at the top of this file (originally taken from the `ffi_support` crate). It was in reference to implementing a clone from `bool` to `bool`, but converting `i32` or `u32` to `bool` is fine.

Comment (lionel1704) [reply src/repr_c.rs]: No particular reason for using i32.
I'll change my implementation.

**Pull request merged**